pkg/stanza/entry/copy.go

```diff
@@ -3,71 +3,52 @@
 
 package entry // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/entry"
 
-import "encoding/json"
+import (
+	"bytes"
+	"encoding/json"
+	"maps"
+	"slices"
+)
 
 // copyValue will deep copy a value based on its type.
 func copyValue(v any) any {
 	switch value := v.(type) {
 	case string, int, bool, byte, nil:
 		return value
 	case map[string]string:
-		return copyStringMap(value)
+		return maps.Clone(value)
 	case map[string]any:
-		return copyInterfaceMap(value)
+		return copyAnyMap(value)
 	case []string:
-		return copyStringArray(value)
+		return slices.Clone(value)
 	case []byte:
-		return copyByteArray(value)
+		return bytes.Clone(value)
 	case []int:
-		return copyIntArray(value)
+		return slices.Clone(value)
 	case []any:
-		return copyInterfaceArray(value)
+		return copyAnySlice(value)
 	default:
 		return copyUnknown(value)
 	}
 }
 
-// copyStringMap will deep copy a map of strings.
-func copyStringMap(m map[string]string) map[string]string {
-	mapCopy := make(map[string]string)
-	for k, v := range m {
-		mapCopy[k] = v
+// copyAnyMap will deep copy a map of interfaces.
+func copyAnyMap(m map[string]any) map[string]any {
+	if m == nil {
+		return nil
```

Comment:
Should we return empty map here, to make this backward compatible with `copyStringMap`?
```suggestion
		return map[string]string{}
```